Options.py

```diff
@@ -740,7 +740,51 @@ def __str__(self) -> str:
 
     @staticmethod
     def triangular(lower: int, end: int, tri: typing.Optional[int] = None) -> int:
-        return int(round(random.triangular(lower, end, tri), 0))
+        if lower == end:
+            return lower
```

Comment:
This early return should go below the exception so that `tri < lower` or `tri > end` always raises an Exception, even when `lower == end`. I'll move it tomorrow.